lib/list.g

```diff
@@ -369,14 +370,40 @@ DeclareCategoryKernel( "IsRange",
 ##
 #R  IsRangeRep( <obj> )
 ##
+##  <#GAPDoc Label="IsRangeRep">
 ##  <ManSection>
 ##  <Filt Name="IsRangeRep" Arg='obj' Type='Representation'/>
 ##
 ##  <Description>
-##  For internally represented ranges, there is a special representation
-##  which requires only a small amount of memory.
+##  Tests whether <A>obj</A> is represented as a range,
+##  that is by internally storing only the first value, the in- or decrement,
+##  and the last value of the range.
+##  <P/>
+##  To test whether a list is an arithmetic pro- or degression in the
+##  mathematical sense see <Ref Filt="IsRange"/>.
+##  <P/>
+##  Lists created by the syntactic construct
+##  <C>[ <A>first</A>, <A>second</A>  .. <A>last</A> ]</C>,
+##  see <Ref Sect="Ranges"/> , are in <Ref Filt="IsRangeRep"/>.
+##  <P/>
+##  Note that if you change <Ref Filt="IsRangeRep"/> object by assignment,
+##  <Ref Oper="Add"/> or <Ref Oper="Append"/>,
+##  the range will be converted into a plain list, even if the resulting list
```

Comment:
```suggestion
##  the range may be converted into a plain list, even if the resulting list
```
If we write "will", then we essentially promise this will always be the case, thus prevent any future changes to this behavior.

Reply:
Unfortunately this already was stated in the documentation of `ConvertToRangeRep`.
```
##  If you change a mutable range that is represented in the compact way,
##  by assignment, <Ref Oper="Add"/> or <Ref Oper="Append"/>,
##  the range will be converted to the ordinary representation, even if the
##  change is such that the resulting list is still a proper range.
```

Reply:
Granted, it was not very discoverable there. So should we still change the wording here (and also in the `doc/ref/list.xml`) to say "may"?

Reply:
I'd prefer to change it to `may` in both places. But perhaps others (@stevelinton @ChrisJefferson ...) have differing views.

Reply:
@wilfwilson @fingolfin Nobody spoke up against it, so I'll change it.